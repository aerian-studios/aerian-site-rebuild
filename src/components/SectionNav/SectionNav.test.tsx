import * as React from "react";
import * as renderer from "react-test-renderer";
import { SectionNav } from ".";

const keys = {
    caseStudy: "Introduction",
    gallery: "Gallery",
    challenge: "The Challenge",
    solution: "The Solution",
    results: "The Result"
};

describe("SectionNav", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(
                <SectionNav
                    className="myClass"
                    onNavigation={() => false}
                    keyConsts={keys}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders with the correct nav items", () => {
        const tree = renderer
            .create(
                <SectionNav
                    className="myClass"
                    onNavigation={() => false}
                    keyConsts={keys}
                />
            )
            .toJSON();

        expect(
            (tree!.children![0] as renderer.ReactTestRendererJSON).children!
                .length
        ).toEqual(Object.keys(keys).length);
    });
});
