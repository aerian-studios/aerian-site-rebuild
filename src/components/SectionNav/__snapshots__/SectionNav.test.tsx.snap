// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionNav renders correctly 1`] = `
<nav
  className=" myClass"
/>
`;
